File: DSA/algorithms/Star_Patterns/java/Pattern.java
/*

    pyramid reverse
    author : Sameer Saini

 */

import java.util.Scanner;

public class Pattern {
    static Scanner in = new Scanner(System.in);
    public static void main(String args[]){

        // asks for reverse pyramid size
        System.out.print("enter : ");
        int var = in.nextInt();
        
        // starts construction
        for(int x=0;x<var;x++){
            for(int y=0;y<var-x;y++){
                System.out.print(" ");
            }
            for(int y=x;y>0;y--){
                System.out.print("*");
            }
            System.out.print("\n");
        }
        // end of construction
    }

}

File: DSA/algorithms/Star_Patterns/python/pattern.py
"""
    author: sameer saini
    reverse Dynamic Triangle

"""

n = int(input("Enter size of triangle: ")) # takes the size of the triangle as input

for x in range(1,n):
    sam =[" "*int(n-(int(x))),"o"*int(x)] # prints spaces before "o" to make the triangle reverse
    print ("".join(sam)) # prints it one by one
